test/sdk-execution-tests/sdk_execution_tests.py

```diff
@@ -104,16 +106,14 @@ def get_kfp_package_path() -> str:
     return path
 
 
-partial_component_decorator = functools.partial(
+dsl.component = functools.partial(
```

Comment:
Note for reviewer: The existing mocking behavior wasn't working correctly. This is simpler and works, allowing us to test the `executor_main.py` code from each PR's HEAD.